### PR TITLE
Bug 1215716 - urlencode parameters passed through in the query string

### PR DESCRIPTION
**Pull request by petemoore**
Only changed `codegenerator/model/api.go` and  `src/main/java/org/mozilla/taskcluster/client/TaskClusterRequestHandler.java` - the other files get generated.

Therefore please review only the changes to these two files and check that the generated files look sane. Luckily these two files should be the first two files shown in the "Files changed" tab. :) Thanks!

### CODE REVIEW

File: src/main/java/org/mozilla/taskcluster/client/TaskClusterRequestHandler.java [hunk 1]
```diff
@@ -165,4 +167,18 @@ public TaskClusterRequestHandler setBaseURL(String baseURL) {
...
+    /**
+     * Utility method to uri encode a parameter for using in url
+     */
+    protected static String uriEncode(String param) {
+        try {
+            return URLEncoder.encode(param, "UTF-8");
+        } catch (UnsupportedEncodingException e) {
+            // This should not be possible since we specify UTF-8
+            e.printStackTrace();
+            System.exit(64);
```
Comment (jonasfj) [inline src/main/java/org/mozilla/taskcluster/client/TaskClusterRequestHandler.java]: I trust your java skills that this is true... but it sounds right... Just sad they didn't do an `URLEncoder.encodeUTF8(string)` method that couldn't throw..

**Comment by jonasfj** [comment]
`r+`, if you think this is right... I think the encoding might be a bit off:
http://stackoverflow.com/questions/607176/java-equivalent-to-javascripts-encodeuricomponent-that-produces-identical-outpu

Lol, someone uses script `ScriptEngine` to call `encodeURIComponent` rather than just learning to read the bytes of a UTF-8 string and converting anything that doesn't match `[-a-zA-Z0-9._*~'()!]` to hex...

File: src/main/java/org/mozilla/taskcluster/client/TaskClusterRequestHandler.java (hunk 1 above)
Comment (petemoore) [inline src/main/java/org/mozilla/taskcluster/client/TaskClusterRequestHandler.java]: Yeah, kinda sucks hey!

Here are the docs: http://docs.oracle.com/javase/8/docs/api/java/net/URLEncoder.html#encode-java.lang.String-java.lang.String-

**Comment by petemoore** [comment]
My brain is a little frazzled reading through this - i guess it also depends on how taskcluster-base interprets the requests too, i.e. if a(b(x)) will always return x where x is the original sequence of bytes passed into the string parameter of the java client, b is my uriEncode function, and a is the taskcluster-base decode function.

I'm not even sure if the java strings are backed by arbitrary byte sequences, or whether they are guaranteed to contain only valid UTF-8 characters.

If it is ok with you, I'd propose we merge as is, since java client is not used by anything, and if someone starts using it and finds a problem, we can invest time to create unit tests for this to make sure it is 100% ok.

**Comment by jonasfj** [comment]
> If it is ok with you, I'd propose we merge as is, since java client is not used by anything,
++
I can't say I care for the java clint :)

**Pull request merged**